src/common/SearchService/SearchService.js

```diff
@@ -0,0 +1,41 @@
+// @ts-check
+import TrieSearch from 'trie-search';
+
+/**
+ * SearchService - Responsible for all the search operations
+*/
+class SearchService{
+    /**
+     * IData - The data that is to be searched into
+     * @typedef {Object[]} IData
+    */
+    
+    /**
+     * Instantiate a new SearchService
+     * @param {IData} data 
+     */
+    constructor(data){
+        /**
+         * Creates an array of object having two properties
+         * keywords - joint string of all the values in dataRow
+         * dataRow - object
+         * @type {Array<{keywords: string, dataRow: IData}>}
+        */
+        const keywordRowPairs = data.map((dataRow) => ({ keywords: Object.values(dataRow).join(" "), dataRow }));
+        const trie = new TrieSearch("keywords");
+        trie.addAll(keywordRowPairs);
+        this.trie = trie;
+    }
+
+    /**
+     * Searches a keyword in the trie, against the set of joint string of all the attributes
+     * @param {string} keyword 
+     * @returns {IData} result
+    */
+    search(keyword){
+        const result = this.trie.search(keyword);
```

Comment:
We may want to avoid including the word `trie` in the description to allow for changes in the underlying data structure. Additionally, it's good to include descriptions of the parameters and return value to ensure clarity for the user. Finally, the user might be searching with multiple keywords so this should be reflected in the input name and description.   
```suggestion
     * Searches for keywords against the set of joint string of all the attributes
     * @param {string} keywords - the search keywords
     * @returns {IData} result - the resulting rows containing all search keywords
     */
    search(keywords) {
      const result = this.trie.search(keywords);
```